src/devices/acurite.c

```diff
@@ -713,36 +714,103 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
         // M = Message type
         message_type = bb[2] & 0x3f;
 
-        // tower sensor messages are 7 bytes.
-        // TODO: - see if there is a type in the message that
-        // can be used instead of length to determine type
+        // Multiple AcuRite sensors use the same basic message format,
+        // with a shared header and integrity scheme. message type is packed
+        // as the last 3 bits of the 3rd byte (bb[2]).
         if (browlen == ACURITE_TXR_BITLEN / 8) {
+            // Verify parity bits
+            // Bytes 2, 3, 4, and 5 should all have a parity bit in their MSB
+            int parity = parity_bytes(&bb[2], 4);
+            if (parity) {
+                if (decoder->verbose)
+                    bitrow_printf(bb, 7 * 8, "%s: bad parity: ", __func__);
+                continue; // return DECODE_FAIL_MIC;
+            }
+
+            // Verify checksum, add with carry
+            int chk = add_bytes(bb, 6);
+            if ((chk & 0xff) != bb[6]) {
+                if (decoder->verbose)
+                    bitrow_printf(bb, 7 * 8, "%s: bad checksum: ", __func__);
+                continue; // return DECODE_FAIL_MIC;
+            }
```

Comment:
Isn't this checksum check redundant with the one on line 694? If I understand the code correctly, we will never get this far if the first checksum fails, and they appear to calculate over the exact same range and compare the exact same byte.

Reply:
good point. I had seen it (and the previously extant parity check on 693) when first reading through this code, but assumed they were non-enforcing. The existing one is more robust anyways. I'll remove the one I added.

Reply:
The existing parity code is non-enforcing, really just prints the parity, and I am not sure it is that meaningful. But the checksum check was enforcing/correct from what I can tell.